Dart/src/com/jetbrains/lang/dart/ide/marker/DartMethodLineMarkerProvider.java

```diff
@@ -0,0 +1,73 @@
+package com.jetbrains.lang.dart.ide.marker;
+
+import com.intellij.codeHighlighting.Pass;
+import com.intellij.codeInsight.daemon.DaemonCodeAnalyzerSettings;
+import com.intellij.codeInsight.daemon.LineMarkerInfo;
+import com.intellij.codeInsight.daemon.LineMarkerProviderDescriptor;
+import com.intellij.openapi.editor.colors.CodeInsightColors;
+import com.intellij.openapi.editor.colors.EditorColorsManager;
+import com.intellij.openapi.editor.colors.EditorColorsScheme;
+import com.intellij.openapi.editor.markup.GutterIconRenderer;
+import com.intellij.openapi.editor.markup.SeparatorPlacement;
+import com.intellij.psi.PsiComment;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiWhiteSpace;
+import com.intellij.util.FunctionUtil;
+import com.jetbrains.lang.dart.psi.*;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collection;
+import java.util.List;
+
+public class DartMethodLineMarkerProvider extends LineMarkerProviderDescriptor {
```

Comment:
It is enough to implement `LineMarkerProvider`. Extending `LineMarkerProviderDescriptor` makes one more check box to appear in Preferences | Editor | General | Gutter Icons, this is not what we need for method separators as there's already an option for it.